src/client.js

```diff
@@ -1848,6 +1848,10 @@ function SugarApi(args) {
                     // this is not our message, ignore it
                     return;
                 }
+                // ignore messages from yiu3/timer.js
```

Comment:
yui3.

Could you please explain how did we got trigger of `postMessage` by yui3 means? As fas as I know this is done in Mango's `sugarcrm/modules/Users/tpls/AuthenticateParent.tpl` and nowhere else.

Generally what bothers me is the specific (yui3 lib) check in a generic library (given that I absolutely understand that ventana has emerged from Mango code base and somewhat deeply connected w/ it).

Reply:
I received this event with this data from sidecar ping API.
I absolutely agree about ventana, it has deeply connected with sidecar

Reply:
The issue appears to be more about not checking for an explicit key that this event was meant for this handler.
Checking if `setImmediate` is present max fix messages generated from YUI, but it won't fix other related issues from other libraries that may post a message. 

Right now we do `var authData = $.parseJSON(event.originalEvent.data);` and assume that if the response is falsey, the auth failed. Really we need assume that if the JSON parse fails or if authData is missing a key, like `authData.eventType === 'auth'`. It would require modifying the format of the expected event to probably look more like
```js
{
    eventType:'auth',
    authData: {
        access_token: '...', 
        etc:'...' 
    }
}
```
The upside is that it *should* solve this issue for other possible event sources going forward. (Like custom views or MFE's)